File: partiql-ast/src/main/java/org/partiql/ast/v1/expr/ExprIon.java
package org.partiql.ast.v1.expr;

import org.jetbrains.annotations.NotNull;
import org.partiql.ast.v1.AstNode;
import org.partiql.ast.v1.AstVisitor;
import com.amazon.ionelement.api.IonElement;

import java.util.Collections;
import java.util.Collection;

/**
 * TODO docs, equals, hashcode
 */
public class ExprIon extends Expr {
    @NotNull
    public IonElement value;

    public ExprIon(@NotNull IonElement value) {
        this.value = value;
    }

    @Override
    @NotNull
    public Collection<AstNode> children() {
        return Collections.emptyList();
    }

    @Override
    public <R, C> R accept(@NotNull AstVisitor<R, C> visitor, C ctx) {
        return visitor.visitExprIon(this, ctx);
    }
}

File: partiql-ast/src/main/kotlin/org/partiql/ast/v1/expr/ExprIon.kt
package org.partiql.ast.v1.expr

import com.amazon.ionelement.api.IonElement
import org.partiql.ast.v1.AstVisitor

/**
 * TODO docs, equals, hashcode
 */
public class ExprIon(
    @JvmField
    public var `value`: IonElement,
) : Expr() {
    public override fun children(): List<Expr> = emptyList()

    public override fun <R, C> accept(visitor: AstVisitor<R, C>, ctx: C): R =
        visitor.visitExprIon(this, ctx)
}
